### PR TITLE
Add section about inspection

### PR DESCRIPTION
**Pull request by lentzi90**
Based on these:
- https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/inspection-api.md
- https://github.com/metal3-io/baremetal-operator/blob/main/docs/inspectAnnotation.md

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

File: docs/user-guide/src/bmo/inspect_annotation.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Inspect annotation
+
+The inspect annotation can be used to request the baremetal operator to (re-)inspect a `Ready` BareMetalHost.
+This is useful in case there were hardware changes for example.
+Note that it is only possible to do this when BareMetalHost is in `Ready` state.
+If an inspection request is made while BareMetalHost is any other state than Ready, the request will be ignored.
+
```
Comment (furkatgofurov7) [inline docs/user-guide/src/bmo/inspect_annotation.md:7]: IMO the below deserves "How to use inspect annotaion" header?

Comment (lentzi90) [reply docs/user-guide/src/bmo/inspect_annotation.md:7]: Sure I can add it, but we don't have any sub-sections on the other pages, like reboot annotation :thinking:

Comment (furkatgofurov7) [reply docs/user-guide/src/bmo/inspect_annotation.md:7]: Good point, it is mixed actually, some don't have it and some does (secure boot, live ISO). Let's maybe align all of those in a follow-up PR as discussed offline.

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks @lentzi90!

/lgtm

**Review by furkatgofurov7 — APPROVED** [review]
/approve

**Pull request merged**